src/api/fileApi.ts

```diff
@@ -0,0 +1,87 @@
+import { AwsS3Part } from '@uppy/aws-s3-multipart';
+import { UppyFile } from '@uppy/core';
+import { API_PATHS } from '../utils/constants';
+import { createResource } from './resourceApi';
+import { setAxiosDefaults } from '../utils/axios-config';
+import { ResourceCreationType } from '../types/resource.types';
+import { authenticatedApiRequest } from './api';
+
+setAxiosDefaults();
+
+export enum FileApiPaths {
+  ABORT = '/upload/multipart/uppy/abort',
+  COMPLETE = '/upload/multipart/uppy/complete',
+  CREATE = '/upload/multipart/uppy/create',
+  LIST_PARTS = '/upload/multipart/uppy/listparts',
+  PREPARE = '/upload/multipart/uppy/prepare',
+}
+
+export const abortMultipartUpload = async (uploadId: string, key: string) => {
+  const data = `uploadId=${uploadId}&key=${key}`;
+  const response = await authenticatedApiRequest({
+    url: encodeURI(`${API_PATHS.guiBackendResourcesContentPath}${FileApiPaths.ABORT}`),
+    method: 'POST',
+    data: data,
+  });
+  return response.data;
+};
+
+export const completeMultipartUpload = async (uploadId: string, key: string, parts: AwsS3Part[]) => {
+  const data = `uploadId=${uploadId}&key=${key}&parts=${JSON.stringify(parts)}`;
+  const response = await authenticatedApiRequest({
+    url: encodeURI(`${API_PATHS.guiBackendResourcesContentPath}${FileApiPaths.COMPLETE}`),
+    method: 'POST',
+    data: data,
+  });
+  return response.data;
+};
+export const createMultipartUpload = async (file: UppyFile) => {
+  const data = `filename=${file.name}&size=${file.data.size}&lastmodified=${
+    (file.data as File).lastModified
+  }&mimetype=${file.data.type}`;
+  const createMultipartUploadResponse = await authenticatedApiRequest({
+    url: encodeURI(`${API_PATHS.guiBackendResourcesContentPath}${FileApiPaths.CREATE}`),
+    method: 'POST',
+    data: data,
+  });
+  return createMultipartUploadResponse.data;
+};
+
+export interface createResourceAndMultipartUploadResponseType {
+  resourceIdentifier: string;
+  data: any;
```

Comment:
data har ingen type?
når vi gikk over til strengere linting fikk vi warning på all bruk av `any`, men vi droppa den regelen siden vi måtte skrive om så mye for at den skulle bli blid... du kan se om det går med `unknown` i stedet, for det liker den visst bedre

Reply:
takk for tips. men her hadde jeg egentlig tenkt å lage en type, ja

Reply:
Ettersom det er uppy-stuff - fjerner jeg hele response-typen